subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2545,72 +2560,69 @@ int lwm2m_write_handler(struct lwm2m_engine_obj_inst *obj_inst,
 			ret = lwm2m_write_handler_opaque(obj_inst, res,
 							 res_inst, msg,
 							 data_ptr, data_len);
-			if (ret < 0) {
-				return ret;
-			}
-
 			len = ret;
 			break;
 
 		case LWM2M_RES_TYPE_STRING:
-			engine_get_string(&msg->in, write_buf, write_buf_len);
+			ret = engine_get_string(&msg->in, write_buf,
+						write_buf_len);
 			len = strlen((char *)write_buf);
 			break;
 
 		case LWM2M_RES_TYPE_U32:
 		case LWM2M_RES_TYPE_TIME:
-			engine_get_s64(&msg->in, &temp64);
+			ret = engine_get_s64(&msg->in, &temp64);
 			*(uint32_t *)write_buf = temp64;
```

Comment:
This seems bad, do we really want to write the data to `write_buf` without checking `ret` first to make sure we didn't get any errors? Similarly for all the other types.

Reply:
You're right, this isn't right to write the value in case of error, I've added additional error check before write here and in other cases.